performance profiling with craft cms

dsteinel opened the pull request:
This is a new blog post about Performance Profiling with CraftCMS

It is a introduction to the necessary parts but no deep dive.

jankohofmann commented on _posts/2018-07-23-performance-profiling-with-craft-cms.md:
```diff
@@ -0,0 +1,117 @@
+mainImageAlt: ""
+--- 
+
+Are you encountering slow page-speed with CraftCMS? This guide will help you to identify the bottlenecks of your code. 
```
Maybe start with a short introduction about what Craft CMS is and where / why we use it?

jankohofmann commented on _posts/2018-07-23-performance-profiling-with-craft-cms.md:
```diff
@@ -0,0 +1,117 @@
+Are you encountering slow page-speed with CraftCMS? This guide will help you to identify the bottlenecks of your code. 
+
+### Yii Debug Toolbar
+Craft can help you to debug your templates with the build-in `Yii Debug Toolbar`.  This toolbar provides you with valuable informations about the database queries executed by the page. You can filter or sort all queries to see which one takes more time than expected.
```
valuable information (no "s")

jankohofmann commented on _posts/2018-07-23-performance-profiling-with-craft-cms.md:
```diff
@@ -0,0 +1,117 @@
+### Yii Debug Toolbar
+Craft can help you to debug your templates with the build-in `Yii Debug Toolbar`.  This toolbar provides you with valuable informations about the database queries executed by the page. You can filter or sort all queries to see which one takes more time than expected.
+<br><br>
+> The [Yii Debug Toolbar](https://www.yiiframework.com/extension/yii-debug-toolbar) is a configurable set of panels that display various debug information about the current request/response and when clicked, display more details about the panel's content.
```
displays more details

jankohofmann commented on _posts/2018-07-23-performance-profiling-with-craft-cms.md:
```diff
@@ -0,0 +1,117 @@
+If your code is not split up like in the example above or if you need to profile your twig template more in detail, check out the [craft-twigprofiler](https://github.com/nystudio107/craft-twigprofiler). With this tool you can test smaller chunks of your template.
+
+### Reduce database queries with Eager Loading
+To keep the database queries as low as possible, you need to bundle the queries into bigger pieces and fetch as many related datas as you can at once.  In order to tell Craft in advance which data we are going to use, we need to **Eager Load** them.
```
related data (no "s")

jankohofmann commented on _posts/2018-07-23-performance-profiling-with-craft-cms.md:
```diff
@@ -0,0 +1,117 @@
+This example shows you how you Eager Load an imageField:
+
+#### No Eager Loading
+{% highlight twig %}
```
you might want to format the whole block as a code snippet (with the three backticks)

dsteinel replied:
This is actually how you highlight code snippets with jekyll. It also works with backticks though.

jankohofmann replied:
Uh, right. Forgot that the dev blog runs on Jekyll. Thought it was generic Markdown. My bad 😇

jankohofmann commented on _posts/2018-07-23-performance-profiling-with-craft-cms.md:
```diff
@@ -0,0 +1,117 @@
+{% endhighlight %}
+
+#### With Eager Loading
+{% highlight twig %}
```
same here (code formatting)

jankohofmann commented on _posts/2018-07-23-performance-profiling-with-craft-cms.md:
```diff
@@ -0,0 +1,117 @@
+* If you want to dive deeper, here is a really good & detailed read about [Eager loading](https://nystudio107.com/blog/speed-up-your-craft-cms-templates-with-eager-loading).
+
+### Conclusion
+Well of course this heavily depends on the page you are serving. But in general it is a good idea to have  the first contentful paint (FCP) below 2s. If you are higher than that, does not mean you are serving a bad website, but this means that your user will lose interest very quickly. As you can see in the illustration, the bounce rate increases slowly after 2 seconds and rapidly above 3 seconds ...
```
Of course the overall performance heavily depends on the page you are serving, but in general...

jankohofmann commented on _posts/2018-07-23-performance-profiling-with-craft-cms.md:
```diff
@@ -0,0 +1,117 @@
+After you optimised your template, the next step would be to enable caching. Craft has its build-in [caching tag](https://docs.craftcms.com/v3/dev/tags/cache.html) with which you can cache specific parts of your template.
+The [Blitz Plugin](https://github.com/putyourlightson/craft-blitz) is a intelligent static file caching which will improve the first contentful paint even more.
+
+Do not only check your page-speed with the Craft debugger. Also check the network tab in your [browser inspector](https://developers.google.com/web/tools/chrome-devtools/network-performance/) and also use tools like these one:
```
I would love to read an example with concrete figures how this has improved page speed in one of our projects (Friday Magazine for instance)

jankohofmann commented on _posts/2018-07-23-performance-profiling-with-craft-cms.md:
```diff
@@ -0,0 +1,117 @@
+<br>
+* You buy it
+* You run it locally or on a test environment. [Here is how](https://craftcms.com/support/try-craft-pro)
+![Enable Craft Pro Vesion](http://res.cloudinary.com/dsteinel/image/upload/c_scale,w_512/v1532351330/Espi/blog/performance%20profiling/perfomance-profiling__craft-pro-version.png)
```
maybe we should specify the image sources somewhere? (looks like you recorded them yourself?)

dsteinel replied:
Yeah I did not figured out how we can use the local path. And I fell in love with cloudinary (งツ)ว.
What would be your idea where to host them? Maybe company AWS?

dsteinel commented:
@jankohofmann thanks a lot for the nice feedback! To make it more real worldy (¯\_(ツ)_/¯) is a very good idea!
 
Changes I did:
- Modified a new intro section to make it more ESPI real
- Added a real world example
- Modified the conclusion part

@edenspiekermann/devs  fresh update 🎈

dsteinel commented:
👀 @edenspiekermann/all could someone have another look 👀
